src/main/java/jotato/quantumflux/items/ItemMagnet.java

```diff
@@ -67,12 +69,14 @@ public void onUpdate(ItemStack item, World world, Entity entity, int i, boolean
 			itemToGet.delayBeforeCanPickup = 50;
 
 			EntityItemPickupEvent pickupEvent = new EntityItemPickupEvent(player, itemToGet);
+			ItemPickupEvent itemPickupEvent = new ItemPickupEvent(player, itemToGet);
 			MinecraftForge.EVENT_BUS.post(pickupEvent);
+			FMLCommonHandler.instance().bus().post(itemPickupEvent);
 			ItemStack itemStackToGet = itemToGet.getEntityItem();
 			int stackSize = itemStackToGet.stackSize;
 			
 			
-			if (pickupEvent.getResult() == Result.ALLOW || stackSize <= 0 || player.inventory.addItemStackToInventory(itemStackToGet))
+			if (pickupEvent.getResult() == Result.ALLOW || stackSize <= 0 || itemPickupEvent.pickedUp == itemToGet || player.inventory.addItemStackToInventory(itemStackToGet))
```

Comment:
I need to think about this for a bit. If another mod subscribes to the ItemPickupEvent and changes which item was picked up, then this wouldn't return true.  My concern would be possible item duplication...i think

Granted, I can't think of a reason WHY a mod would do that, but look at the EntityItemNetherbane. I do something similar to a diamond sword that is dropped.